astropy/io/fits/fitsrec.py

```diff
@@ -1307,9 +1305,7 @@ def _scale_back_ascii(self, col_idx, input_field, output_field):
 
         if lead < 0:
             warnings.warn(
-                "Column {!r} starting point overlaps the previous column.".format(
-                    col_idx + 1
-                )
+                f"Column {col_idx + 1!r} starting point overlaps the previous column."
```

Comment:
`!r` does nothing for integers (not your fault obviously...)

Reply:
Done here and in following almost identical message.